File: 2015/AI/AIClient_JAVA/src/event/MineHitEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** MinehitEvent.java
 ** Event that happen when a character hit a mine
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

import world.World;

public class MineHitEvent extends IngoingEvent {

	private int hitTeamId;
	private int hitCharacterId;
	private int originTeamId;
	private int originCharacterId;

	@Override
	public void execute() {
		World.getInstance().getTeam(hitTeamId).getCharacter(hitCharacterId).hitByMine();
		World.getInstance().getTeam(originTeamId).getCharacter(originCharacterId).mineHit();
	}

	@Override
	public boolean fillArguments(String message) {
		String[] messageParts = message.split(SEPARATOR, 5);
		
		hitTeamId = convertCharToNumeral(messageParts[0]);
		if (hitTeamId == 0) {
			return false;
		}
		hitCharacterId = convertCharToNumeral(messageParts[1]);

		originTeamId = convertCharToNumeral(messageParts[2]);
		if (originTeamId == 0) {
			return false;
		}
		originCharacterId = convertCharToNumeral(messageParts[3]);
		
		return true;
	}

}

File: 2015/AI/AIClient_Python/src/event/MineHitEvent.py
'''
Created on Jan 2, 2015

@author: scarriere
'''
from event.IngoingEvent import IngoingEvent
from aiclient.Singleton import Singleton
from world.World import World

class MineHitEvent(IngoingEvent):

    def __init__(self):
        self.hitTeamId = 0
        self.hitCharacterId = 0
        self.originTeamId = 0
        self.originCharacterId = 0
    
    def fillArguments(self, string):
        arguments = string.split(":", 4)
        
        self.hitTeamId = int(arguments[0])
        self.hitCharacterId = int(arguments[1])
        self.originTeamId = int(arguments[2])
        self.originCharacterId = int(arguments[3])
        
    def execute(self):
        world = Singleton(World)
        world.getTeam(self.hitTeamId).characters[self.hitCharacterId]._hitByMine()
        world.getTeam(self.originTeamId).characters[self.originCharacterId]._mineHit()